### ISSUE TITLE
[ENH]: Cannot disable coordinate display in ToolManager/Toolbar (it's doable in NavigationToolbar2)

### ISSUE THREAD

**Issue by positron96** [matplotlib/matplotlib#28985]
### Problem

When creating NavigationToolbar2, you can disable coordinates display with a simple constructor argument (at least in Qt backend):
`navbar = NavigationToolbar2QT(canvas, parent_widget, coordinates=False)`

There doesn't seem to be a similar feature in ToolManager API. Judging from the code, its message event will unconditionally be propagated and drawn on the toolbar.

### Proposed solution

Port this feature from NavigationToolbar2 to ToolManager

**Comment by anntzer** [comment]
I think the following works? `figure.canvas.manager.toolmanager.remove_tool("position")`

**Comment by positron96** [comment]
Hi, that was by thought also. But for me it causes an exception inside backend_qt.ToolbarQt.remove_item:

```
  File ".venv\Lib\site-packages\matplotlib\cbook.py", line 298, in process
    func(*args, **kwargs)
  File ".venv\Lib\site-packages\matplotlib\backend_bases.py", line 3249, in <lambda>
    lambda event: self.remove_toolitem(event.tool.name))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".venv\Lib\site-packages\matplotlib\backends\backend_qt.py", line 964, in remove_toolitem    
    for button, handler in self._toolitems[name]:
                           ~~~~~~~~~~~~~~~^^^^^^
KeyError: 'position'
```

I assume it wants to remove a button for the tool,  but position tool works differently via a QLabel instead.

**Comment by tacaswell** [comment]
It looks like this was changed in 3d0f15391ceeda16a1e928d3b6667cf6b5c1c276 but I think it will still fail with an index error on the unpacking.

**Comment by anntzer** [comment]
I cannot actually reproduce the issue, whether as of master or before the commit mentioned by @tacaswell; the code I posted actually works for me.  A reproducing code would be needed to move forward...

**Comment by positron96** [comment]
Here is a minimal failing program:
```
import matplotlib
matplotlib.use('qt5agg')
matplotlib.rcParams['toolbar'] = 'toolmanager'
import matplotlib.pyplot as plt

print(matplotlib.__version__)  # 3.8.4

fig = plt.figure()
plt.plot([1, 2, 3])

fig.canvas.manager.toolmanager.remove_tool('position')

plt.show()

```

Interestingly, it raises the above `KeyError: 'position'`, but continues to show the plot.

**Comment by anntzer** [comment]
OK, I can confirm that this was indeed fixed by 3d0f1539 (so not released yet), not sure why I failed to see that on my first attempt.
